src/os/bluestore/BlueStore.cc

```diff
@@ -7664,17 +7664,9 @@ void BlueStore::_kv_sync_thread()
       // can rely on the bluefs commit to flush the device and make
       // deferred aios stable.  that means that if we do have done deferred
       // txcs AND we are not on a single device, we need to force a flush.
-      if (bluefs_single_shared_device && bluefs) {
-	if (num_aios) {
-	  force_flush = true;
-	} else if (kv_committing.empty() && kv_submitting.empty() &&
-	    deferred_stable.empty()) {
-	  force_flush = true;  // there's nothing else to commit!
-	} else if (deferred_aggressive) {
-	  force_flush = true;
-	}
-      } else
+      if (!(bluefs_single_shared_device && bluefs)) {
 	force_flush = true;
+      }
```

Comment:
This isn't correct.  We need to make sure the aio for a set of transactions is stable and durable on the device *before* we commit the metadata.  If we do it all at once we can't ensure that the data is durable before the metadata, and a crash might mean that the metadata made it to disk but the data did not, resulting in corrupted state.